lib/internal/streams/end-of-stream.js

```diff
@@ -67,6 +67,24 @@ function eos(stream, opts, callback) {
     }
   };
 
+  const readableDestroyed = stream._readableState &&
+    stream._readableState.destroyed;
+  const writableDestroyed = stream._writableState &&
+    stream._writableState.destroyed;
+
+  if (readableDestroyed || writableDestroyed) {
+    let err;
+    if (readable) {
+      if (!stream._readableState || !stream._readableState.endEmitted)
+        err = new ERR_STREAM_PREMATURE_CLOSE();
```

Comment:
I’m not convinced this is the right error to be used here. PREMATURE_CLOSE is used for a stream that was terminated “unnaturally”. However in this case, the stream is already ended and we don’t know if it errored or not. Maybe we should not have an error in this case?